### DISCUSSION TITLE
Integrate keycloak to get token

### DISCUSSION THREAD

**Discussion by xiaoxiji1234** [lowdefy/lowdefy#1151] [Questions and Help]
Hi, I used `keycloak` as a user login in lowdefy, calling `REST API` of `keycloak` in lowdefy, but I found lowdefy only saved `idtoken` and how to get user `access_.token` allows me to call `REST API`

**Comment by SamTolmay** [comment]
Hi @xiaoxiji1234 

Unfortunately this is not possible using V3, but the authentication system in V4 is a lot more flexible and will be able to do this.
